utils/s2n_safety.h

```diff
@@ -95,13 +95,20 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD_GOTO( x , label ) do {if ( (x) < 0 ) goto label;} while (0)
 #define GUARD_PTR( x )          do {if ( (x) < 0 ) return NULL;} while (0)
 
+/* Similar to GUARD, but preserves the blocking error code. -2 is left to preserve backwards compatibility .*/
+#define GUARD_AGAIN( x )                                    \
```

Comment:
Does this need to be a new GUARD_* definition? It's only used in one place.

If we do need it, can we name it something more descriptive?

Reply:
It is only used in one place today, but it will be used in more places as the external key server support is added. I figured the name was meant to guard against EAGAIN conditions.

Maybe GUARD_NONBLOCKING() is more descriptive? You can use this guard when you are guarding a nonblocking function call.